url8d3c4d17f5_catseye_JaC64_tgz-pJ8-JaC64J8/src/com/dreamfabric/jac64/C64Screen.java

```diff
@@ -0,0 +1,2025 @@
+      System.out.print("A1");
+    } else {
+      System.out.print("A0");
+    }
+
+    // The c64 has id = 1
+    int sdata = ((iecLines & 0x40) == 0) ? 1 : 0;
+    System.out.print(" C" + sdata);
+    sdata = ((iecLines & 0x80) == 0) ? 1 : 0;
+    System.out.print(" D" + sdata);
+
+    // The 1541 has id = 2
+    sdata = ((c1541Chips.iecLines & 0x40) == 0) ? 1 : 0;
+    System.out.print(" c" + sdata);
+    sdata = ((c1541Chips.iecLines & 0x80) == 0) ? 1 : 0;
+    System.out.print(" d" + sdata);
+
+    System.out.println(" => C" +
+        ((iecLines & c1541Chips.iecLines & 0x80) == 0 ? 1 : 0)
+        + " D" +
+        ((iecLines & c1541Chips.iecLines & 0x40) == 0 ? 1 : 0));
+  }
+
+  private void setVideoMem() {
+    if (VIC_MEM_DEBUG) {
+      monitor.info("setVideoMem() cycles since line: " +
+          (cpu.cycles - lastLine) +
+          " cycles since IRQ: " + (cpu.cycles-lastIRQ) +
+          " at " + vbeam);
+    }
+    // Set-up vars for screen rendering
+    vicBank = (~(~cia2DDRA | cia2PRA) & 3) << 14;
+    charSet = vicBank | (vicMem & 0x0e) << 10;
+    videoMatrix = vicBank | (vicMem & 0xf0) << 6;
+    vicBase = vicBank | (vicMem & 0x08) << 10;
+    spr0BlockSel = 0x03f8 + videoMatrix;
+
+//    monitor.info("--------------------");
+//    monitor.info("0xdd00: 0x"+Integer.toHexString(memory[IO_OFFSET + 0xdd00]));
+//    monitor.info("0xd018: 0x"+Integer.toHexString(memory[IO_OFFSET + 0xd018]));
+//    monitor.info("vicBank: 0x"+Integer.toHexString(vicBank));
+//    monitor.info("videoMatrix: 0x"+Integer.toHexString(videoMatrix-vicBank));
+//    monitor.info("charSet: 0x"+Integer.toHexString(charSet-vicBank));
+
+    //check if vic not looking at char rom 1, 2, 4, 8
+    // This is not correct! Char Rom is not everywhere!!!! - find out!
+    if ( (vicMem & 0x0c) != 4 || (vicBank & 0x4000) == 0x4000) {
+      charMemoryIndex = charSet;
+    } else {
+      charMemoryIndex = (((vicMem & 0x02) == 0) ? 0 : 0x0800) +
+        CPU.CHAR_ROM2;
+    }
+  }
+
+  private void initUpdate() {
+    vc = 0;
+    vcBase = 0;
+    vmli = 0;
+    //     rc = 0;
+    updating = true;
+
+//  // First rendered line will start at cpu.cycles - no at next_scan!
+//  firstLine = nextScanLine;
+
+    for (int i = 0; i < 8; i++) {
+      sprites[i].nextByte = 0;
+      sprites[i].painting = false;
+      sprites[i].spriteReg = 0;
+    }
+
+    if (colors == null) {
+      colors = new Color[16];
+      for (int i = 0; i < 16; i++) {
+        colors[i] = new Color(cbmcolor[i]);
+      }
+    }
+    canvas.setBackground(colors[memory[IO_OFFSET + 0xd020] & 15]);
+  }
+
+  // -------------------------------------------------------------------
+  // Screen rendering!
+  // -------------------------------------------------------------------
+  // keep track of if the border is to be painted...
+  private int borderState = 0;
+  private boolean notVisible = false;
+  private int xPos = 0;
+  private long lastCycle = 0;
+  
+  public final void clock(long cycles) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 292 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1022-1035 into a separate method.